Allows to produce clean test run even when XSLT is configured outside of
source directory.

--- libxslt-1.1.37/tests/REC/Makefile.am
+++ libxslt-1.1.37/tests/REC/Makefile.am
@@ -100,7 +100,7 @@ continue ; \
 	  fi ; \
 	  log=`$(CHECKER) $(top_builddir)/xsltproc/xsltproc \
 	  	$(srcdir)/$$name.xsl $(srcdir)/$$name.xml \
-	  	> $$name.res 2>$$name.bad ; \
+	  	2>&1 > $$name.res | sed 's|$(srcdir)|.|' >$$name.bad ; \
 	  if [ ! -f $(srcdir)/$$name.out ] ; then \
 	  	cp $$name.res $(srcdir)/$$name.out ; \
 		if [ -s $$name.bad ] ; then \
@@ -162,7 +162,7 @@ continue ; \
 	  fi ; \
 	  log=`$(CHECKER) $(top_builddir)/xsltproc/xsltproc --nodict \
 	  	$(srcdir)/$$name.xsl $(srcdir)/$$name.xml \
-	  	> $$name.res 2>$$name.bad ; \
+	  	2>&1 > $$name.res | sed 's|$(srcdir)|.|' >$$name.bad ; \
 	  if [ ! -f $(srcdir)/$$name.out ] ; then \
 	  	cp $$name.res $(srcdir)/$$name.out ; \
 		if [ -s $$name.bad ] ; then \
--- libxslt-1.1.37/tests/documents/Makefile.am
+++ libxslt-1.1.37/tests/documents/Makefile.am
@@ -27,11 +27,13 @@ diff $(srcdir)/test.result result; \
 	grep "MORY ALLO" .memdump  | grep -v "MEMORY ALLOCATED : 0" || true;\
 	rm -f result)
 	@($(CHECKER) $(top_builddir)/xsltproc/xsltproc $(srcdir)/test.xsl $(srcdir)/test_bad.xml > result 2>err ; \
+	sed 's|$(srcdir)/||' err > err.tmp; mv err.tmp err; \
 	diff $(srcdir)/test_bad.result result; \
 	diff $(srcdir)/test_bad.err err; \
 	grep "MORY ALLO" .memdump  | grep -v "MEMORY ALLOCATED : 0" || true;\
 	rm -f result err)
 	@($(CHECKER) $(top_builddir)/xsltproc/xsltproc $(srcdir)/message.xsl $(srcdir)/message.xml > result 2>&1 ; \
+	gsed -i 's|$(srcdir)|.|' result; \
 	diff $(srcdir)/message.result result; \
 	grep "MORY ALLO" .memdump  | grep -v "MEMORY ALLOCATED : 0" || true;\
 	rm -f result)
--- libxslt-1.1.37/tests/exslt/saxon/Makefile.am
+++ libxslt-1.1.37/tests/exslt/saxon/Makefile.am
@@ -22,7 +22,8 @@ @(for i in $(srcdir)/*.xsl ; do \
 	  name=`basename $$i .xsl` ; \
 	  if [ ! -f $(srcdir)/$$name.xml ] ; then continue ; fi ; \
 	  log=`$(CHECKER) $(top_builddir)/xsltproc/xsltproc \
-		$(srcdir)/$$name.xsl $(srcdir)/$$name.xml > $$name.res 2>$$name.bad;\
+		$(srcdir)/$$name.xsl $(srcdir)/$$name.xml \
+	 	   2>&1 > $$name.res | sed 's|$(srcdir)|.|' > $$name.bad; \
 	  if [ ! -f $(srcdir)/$$name.out ] ; then \
 		cp $$name.res $(srcdir)/$$name.out ; \
 		if [ -s $$name.bad ] ; then \
--- libxslt-1.1.37/tests/general/Makefile.am
+++ libxslt-1.1.37/tests/general/Makefile.am
@@ -251,16 +251,20 @@ for j in $(srcdir)/$$doc*.xsl ; do \
 	  if [ ! -f $$j ] ; then continue ; fi ; \
 	  if [ -d $$j ] ; then continue ; fi ; \
 	  name=`basename $$j .xsl`; \
+	  real_docs_path=`cd $(srcdir)/../docs && pwd`; \
 	  out=$(srcdir)/"$$name".out; \
 	  err=$(srcdir)/"$$name".err; \
 	  log=`$(CHECKER) $(top_builddir)/xsltproc/xsltproc --maxdepth 200 $$j $$i \
-	  	> result.$$name 2>err.$$name; \
+	  	2>&1 > result.$$name | sed 's|$(srcdir)|.|' >err.$$name; \
 	  if [ ! -f $$out ] ; then \
 	  	cp result.$$name $$out ; \
 		if [ -s err.$$name ] ; then \
 			cp err.$$name $$err ; \
 		fi ; \
 	  else \
+		grep '../docs/' $$out >/dev/null && \
+		  sed "s|$$real_docs_path|../docs|" result.$$name > \
+		  result.$$name.tmp && mv result.$$name.tmp result.$$name; \
 	  	diff $$out result.$$name; \
 		if [ -s $$err ] ; then \
 	  		diff $$err err.$$name; \
@@ -283,16 +287,20 @@ for j in $(srcdir)/$$doc*.xsl ; do \
 	  if [ ! -f $$j ] ; then continue ; fi ; \
 	  if [ -d $$j ] ; then continue ; fi ; \
 	  name=`basename $$j .xsl`; \
+	  real_docs_path=`cd $(srcdir)/../docs && pwd`; \
 	  out=$(srcdir)/"$$name".out; \
 	  err=$(srcdir)/"$$name".err; \
 	  log=`$(CHECKER) $(top_builddir)/xsltproc/xsltproc --nodict --maxdepth 200 $$j $$i \
-	  	> result.$$name 2>err.$$name; \
+	  	2>&1 > result.$$name | sed 's|$(srcdir)|.|' >err.$$name; \
 	  if [ ! -f $$out ] ; then \
 	  	cp result.$$name $$out ; \
 		if [ -s err.$$name ] ; then \
 			cp err.$$name $$err ; \
 		fi ; \
 	  else \
+		grep '../docs/' $$out >/dev/null && \
+		  sed "s|$$real_docs_path|../docs|" result.$$name > \
+		  result.$$name.tmp && mv result.$$name.tmp result.$$name; \
 	  	diff $$out result.$$name; \
 		if [ -s $$err ] ; then \
 	  		diff $$err err.$$name; \
--- libxslt-1.1.37/tests/plugins/Makefile.am
+++ libxslt-1.1.37/tests/plugins/Makefile.am
@@ -29,8 +29,8 @@ @echo '## Running plugin tests'
 test tests: $(top_builddir)/xsltproc/xsltproc test-logall $(noinst_LTLIBRARIES)
 	@LD_LIBRARY_PATH=$(plugindir):$(top_builddir)/libxslt/.libs:$(LD_LIBRARY_PATH) \
 	 LIBXSLT_PLUGINS_PATH=$(plugindir) \
-	 $(top_builddir)/xsltproc/xsltproc plugin.xsl plugin.xml > plugin.res
-	@diff plugin.out plugin.res
+	 $(top_builddir)/xsltproc/xsltproc $(srcdir)/plugin.xsl $(srcdir)/plugin.xml > plugin.res
+	@diff $(srcdir)/plugin.out plugin.res
 	@rm plugin.res
 
 else
